File: fis-nutricion-infantil/backend/src/main/java/backend/service/PerfilTutorService.java
package backend.service;

import backend.model.PerfilTutor;
import backend.repository.PerfilTutorRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PerfilTutorService {
    private final PerfilTutorRepository perfilTutorRepository;

    public PerfilTutorService(PerfilTutorRepository perfilTutorRepository) {
        this.perfilTutorRepository = perfilTutorRepository;
    }

    public PerfilTutor guardar(PerfilTutor perfilTutor) {
        perfilTutor.crearPerfil();
        return perfilTutorRepository.save(perfilTutor);
    }

    public PerfilTutor actualizar(Integer id, PerfilTutor perfilTutor) {
        if (!perfilTutorRepository.existsById(id)) {
            throw new IllegalArgumentException("Perfil tutor no encontrado.");
        }
        perfilTutor.actualizarDatos();
        return perfilTutorRepository.save(perfilTutor);
    }

    public List<PerfilTutor> listar() {
        return perfilTutorRepository.findAll();
    }

    public Optional<PerfilTutor> buscarPorId(Integer id) {
        return perfilTutorRepository.findById(id);
    }

    public Optional<PerfilTutor> buscarPorCorreo(String correo) {
        return perfilTutorRepository.findByCorreo(correo);
    }

    public void eliminar(Integer id) {
        perfilTutorRepository.deleteById(id);
    }
}

File: fis-nutricion-infantil/frontend/src/services/perfilTutorService.js
import apiClient from './apiClient'

export const perfilTutorService = {
  listar: () =>
    apiClient.get('/perfiles-tutor').then((response) => response.data),

  buscarPorId: (id) =>
    apiClient.get(`/perfiles-tutor/${id}`).then((response) => response.data),

  buscarPorCorreo: (correo) =>
    apiClient
      .get('/perfiles-tutor/buscar', { params: { correo } })
      .then((response) => response.data),

  guardar: (data) =>
    apiClient.post('/perfiles-tutor', data).then((response) => response.data),

  actualizar: (id, data) =>
    apiClient.put(`/perfiles-tutor/${id}`, data).then((response) => response.data),

  eliminar: (id) =>
    apiClient.delete(`/perfiles-tutor/${id}`).then((response) => response.data),
}